mitiq/tests/test_conversions.py

```diff
@@ -176,3 +176,36 @@ def test_atomic_one_to_many_converter(to_type):
     circuits = returns_several_circuits(circuit, return_mitiq=True)
     for circuit in circuits:
         assert isinstance(circuit, cirq.Circuit)
+
+
+def test_noise_scaling_converter_with_qiskit_idle_qubits_and_barriers():
+    """Idle qubits must be preserved even if the input has barriers.
+    Test input:
+         ┌───┐ ░
+    q_0: ┤ X ├─░─
+         └───┘ ░
+    q_1: ──────░─
+         ┌───┐ ░
+    q_2: ┤ X ├─░─
+         └───┘ ░
+    q_3: ────────
+    Expected output:
+         ┌───┐
+    q_0: ┤ X ├
+         └───┘
+    q_1: ─────
+         ┌───┐
+    q_2: ┤ X ├
+         └───┘
+    q_3: ─────
+    """
+    test_circuit_qiskit = qiskit.QuantumCircuit(4)
+    test_circuit_qiskit.x(0)
+    test_circuit_qiskit.x(2)
+    test_circuit_qiskit.barrier(0, 1, 2)
+    scaled = scaling_function(test_circuit_qiskit)
+    # Mitiq expected to remove qiskit barriers
```

Comment:
Actually, if the expected behavior is to remove barriers, isn't it simpler to first remove all barriers before adding identities?

Reply:
Applied this approach in the last commit.